bpf/bpf_host.c

```diff
@@ -595,13 +595,13 @@ handle_ipv4(struct __ctx_buff *ctx, __u32 secctx __maybe_unused,
 			bool __maybe_unused is_dsr = false;
 
 			int ret = nodeport_lb4(ctx, ip4, ETH_HLEN, secctx, ext_err, &is_dsr);
-
+#ifdef ENABLE_IPV6
 			if (ret == NAT_46X64_RECIRC) {
 				ctx_store_meta(ctx, CB_SRC_LABEL, secctx);
 				return tail_call_internal(ctx, CILIUM_CALL_IPV6_FROM_NETDEV,
 							  ext_err);
 			}
-
+#endif
```

Comment:
Curious, why didn't we catch the same case in `bpf_xdp`?

(I would hope that the agent enforces `IPv6=true` when using the NAT46x64 gateway, so this is probably harmless for real deployments)

Reply:
> Curious, why didn't we catch the same case in bpf_xdp?

It could just be that we are not testing an instance where we have IPv4=true, IPv6=false, NodeportAcceleration=true ?